lib/src/client.dart

```diff
@@ -2161,9 +2169,14 @@ class Client extends MatrixApi {
       // stop stale group call checker for left room.
       room.stopStaleCallsChecker(room.id);
 
+      // in order to keep the archive in sync, add left room to archive
+      if (chatUpdate is LeftRoomUpdate) {
```

Comment:
the `membership == Membership.leave` above should be enough

Reply:
No, because then one would need a cast.

Reply:
Unless the type checker can see, that the only way for membership to be leave, is for it to be a LeftRoomUpdate? I don't think it usually sees that, but if it does, the if statement is unnecessary yes.

Reply:
in an ideal world it would, but doesn't look like it does rn. Thanks!